docs/API.md

```diff
@@ -46,48 +67,74 @@ Similar to as `mapProps()`, but child props are only re-computed when one of the
 
 ```js
 withProps(
-  props: Object,
-  BaseComponent: ReactElementType
-): ReactElementType
+  props: Object
+): HigherOrderComponent
 ```
 
 Passes additional props to the base component. Similar to `defaultProps()`, except the provided props take precedence over props from the owner.
 
-### `withAttachedProps()`
+### `withHandlers()`
 
 ```js
-withAttachedProps(
-  createChildProps: (getProps: () => Object) => Object,
-  BaseComponent: ReactElementType
-): ReactElementType
+withHandlers(
+  handlerCreators: {
+    [handlerName: string]: (props: Object) => Function
+  }
+): HigherOrderComponent
 ```
 
-**Note: The name of this helper will likely change to avoid confusion.**
+Takes an object map of handler creators. These are higher-order functions that accept a set of props and return a function handler:
+
+This allows the handler to access the current props via closure, without needing to change its signature.
 
-The first parameter `createChildProps()` is a function which accepts a function `getProps()`, which returns the current owner props. The props returned by `createChildProps()` are immutable and do not change throughout the lifecycle of the component.
+Handlers are passed to the base component as immutable props, whose identities are preserved across renders. This avoids a common pitfall where functional components create handlers inside the body of the function, which results in a new handler on every render and breaks downstream `componentShouldUpdate()` optimizations that rely on prop equality.
+
+Usage example:
+
+```js
+const enhanceForm = compose(
+  withState('value', 'updateValue', ''),
+  withHandlers({
+    onChange: props => event => {
+      props.updateValue(event.target.value)
+    },
+    onSubmit: props => event => {
+      event.preventDefault()
+      submitForm(props.value)
```

Comment:
On the test file you only have the `onChange` handler, which works since the only prop it uses is the `updateValue` function, which never changes. Unless I'm missing something here the onSubmit